libs/api/domains/license-service/src/lib/client/machine-license-client/machineLicenseMappers.ts

```diff
@@ -8,11 +8,27 @@ import {
   GenericUserLicensePayload,
 } from '../../licenceService.type'
 
+import isAfter from 'date-fns/isAfter'
+
 export const parseMachineLicensePayload = (
   license: VinnuvelaDto,
 ): GenericUserLicensePayload | null => {
   if (!license) return null
 
+  let expired: boolean | null = null
+
+  expired = license.vinnuvelaRettindi
+    ? license.vinnuvelaRettindi
+        .filter((field) => field.kenna || field.stjorna)
+        .every(
+          (field: VinnuvelaRettindiDto) =>
+            field.kenna &&
+            !isAfter(new Date(field.kenna), new Date()) &&
+            field.stjorna &&
+            !isAfter(new Date(field.stjorna), new Date()),
+        )
+    : null
```

Comment:
Big enough to warrant refactoring into a function (and add tests?)

Reply:
Moved to function